torchvision/datasets/celeba.py

```diff
@@ -88,23 +92,44 @@ def __init__(
         }
         split_ = split_map[verify_str_arg(split.lower(), "split",
                                           ("train", "valid", "test", "all"))]
+        splits = self._load_csv("list_eval_partition.txt", header=None, index_col=0)
+        identity = self._load_csv("identity_CelebA.txt", header=None, index_col=0)
+        bbox = self._load_csv("list_bbox_celeba.txt", header=1, index_col=0)
+        landmarks_align = self._load_csv("list_landmarks_align_celeba.txt", header=1, index_col=0)
+        attr = self._load_csv("list_attr_celeba.txt", header=1, index_col=0)
+
+        mask = slice(None) if split_ is None else (splits.data == split_).squeeze()
+
+        self.filename = splits.index
+        self.identity = identity.data[mask]
+        self.bbox = bbox.data[mask]
+        self.landmarks_align = landmarks_align.data[mask]
+        self.attr = attr.data[mask]
+        self.attr = (self.attr + 1) // 2  # map from {-1, 1} to {0, 1}
+        self.attr_names = attr.header
+
+    def _load_csv(
+        self,
+        filename: str,
+        header: int = None,
+        index_col: int = None
```

Comment:
Given that all files here use `index_col=0`, I would simply hardcode the behavior and remove the parameter. No need to make this more general as it needs to be.